oracle/oracle.js

```diff
@@ -1,58 +1,60 @@
 require('dotenv').config()
-const cosmosjs = require("@cosmostation/cosmosjs");
+const kava = require('@kava-labs/javascript-sdk');
 const CoinGecko = require('coingecko-api');
 const cron = require('node-cron');
 
-import { postTxKava, getTxKava } from '../common/txs.js';
-import { newMsgPostPrice } from '../common/msg.js';
-import { loadCoinNames } from './utils.js';
+const coinUtils = require('./utils.js');
 
-// Load chain details, credentials
-const mnemonic = process.env.MNEMONIC
-const lcdURL = process.env.LCD_URL
-const chainID = process.env.CHAIN_ID;
 
-// Load params
-const marketIDs = process.env.MARKET_IDS.split(",");
-const coinNames = loadCoinNames(marketIDs)
+var main = async () => {
+    // Load chain details, credentials
+    const mnemonic = process.env.MNEMONIC
+    const lcdURL = process.env.LCD_URL
+    const chainID = process.env.CHAIN_ID;
 
-// Initiate Kava blockchain
-const kava = cosmosjs.network(lcdURL, chainID);
-kava.setBech32MainPrefix("kava");
-kava.setPath("m/44'/459'/0'/0/0");
+    // Load params
+    const marketIDs = process.env.MARKET_IDS.split(",");
+    const coinNames = coinUtils.loadCoinNames(marketIDs)
 
-// Load account credentials
-const address = kava.getAddress(mnemonic);
-const ecpairPriv = kava.getECPairPriv(mnemonic);
-
-// Initiate the CoinGecko API Client
-const CoinGeckoClient = new CoinGecko();
+    // Initiate Kava blockchain
+    client = new kava.KavaClient(lcdURL);
+    client.setWallet(mnemonic);
+    try {
```

Comment:
+1 we should probably add this try-catch directly to `tx.getTx()` on the client - as is it will throw an uncaught error for others too

Reply:
Added in https://github.com/Kava-Labs/javascript-sdk/pull/11